File: src/2864-Maximum-Odd-Binary-Number/maximum_odd_binary_number.cpp
#include <iostream>
#include <string>

class Solution {
public:
    std::string maximumOddBinaryNumber(std::string s) {
        int ones = 0;

        for (char c : s) {
            if (c == '1') {
                ones += 1;
            }
        }

        std::string ans;

        for (size_t i = 0; i < s.length() - 1; i++) {
            if (ones > 1) {
                ans += '1';
                ones -= 1;
            }
            else {
                ans += '0';
            }
        }

        ans += '1';

        return ans;
    }
};

int main() {
    std::string s = "0101";

    Solution solution;

    std::cout << solution.maximumOddBinaryNumber(s); 

    return 0;
}

File: src/2864-Maximum-Odd-Binary-Number/maximum_odd_binary_number.py
class Solution:
    def maximumOddBinaryNumber(self, s: str) -> str:
        ones = s.count('1')
        ans = ''

        for _ in range(len(s) - 1):
            if ones > 1:
                ans += '1'
                ones -= 1
            else:
                ans += '0'
        
        ans += '1'
        return ans



if __name__ == "__main__":
    s:  str = "0101"

    solution = Solution()

    print(solution.maximumOddBinaryNumber(s))
    